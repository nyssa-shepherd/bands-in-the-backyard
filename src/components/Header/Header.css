@import url('https://fonts.googleapis.com/css?family=Oswald:200,300,400,700');

header {
    background-color: #201F22;
    display: flex;
    font-family: Oswald;
    justify-content: space-between;
    height: 120px;
}

h1 {
    color: #A4E3F2;
    display: inline;
    font-size: 40px;
    padding: 5px 20px;
}

span {
    color: #fff;  
    font-weight: 200;
    font-size: 25px; 
}

.nav {
    color: #2C666E;
    font-size: 18px;
    margin: 10px;
    position: relative;
    text-decoration: transparent;
    top: 50px;
}

.nav:hover {
    color: #A4E3F2; 
    border-bottom: #A4E3F2 3px solid;
    font-weight: 700px; 
}
