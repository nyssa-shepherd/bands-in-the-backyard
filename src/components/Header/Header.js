import React from 'react';
import { NavLink, Route } from 'react-router-dom';
import DefaultPage from '../DefaultPage/DefaultPage';
import Home from '../Home/Home.js';
import Artists from '../Artists/Artists.js';
import './Header.css';

const Header = () => {
  return (
    <div>
      <header>
        <h1>Bands <span>in the</span> Backyard</h1>
        <div>
          <NavLink 
            to='/home' 
            activeClassName='active'
            className='home nav'>
            HOME
          </NavLink>
          <NavLink 
            activeClassName='active'
            to='/artists' 
            className='artists nav'>
              ARTISTS
          </NavLink>
        </div>
      </header>
      <div>
        <Route exact path='/home' component={Home} />
        <Route exact path='/artists' component={Artists} />
      </div>
    </div>
  );
};

export default Header;