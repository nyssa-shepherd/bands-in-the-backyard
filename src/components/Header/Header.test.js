import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header.js';

describe('Header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });

});