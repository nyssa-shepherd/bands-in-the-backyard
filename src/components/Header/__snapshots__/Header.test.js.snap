// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="above-header"
>
          <h1>
                    Bands 
                    <span>
                              in the
                    </span>
                     Backyard
          </h1>
          <div>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="home nav"
                              to="/home"
                    >
                              HOME
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="artists nav"
                              to="/artists"
                    >
                              ARTISTS
                    </NavLink>
          </div>
</div>,
        <header />,
        <div>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/home"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/artists"
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Bands 
              <span>
                            in the
              </span>
               Backyard
</h1>,
            <div>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="home nav"
                            to="/home"
              >
                            HOME
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="artists nav"
                            to="/artists"
              >
                            ARTISTS
              </NavLink>
</div>,
          ],
          "className": "above-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Bands ",
                <span>
                  in the
</span>,
                " Backyard",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Bands ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "in the",
                },
                "ref": null,
                "rendered": "in the",
                "type": "span",
              },
              " Backyard",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="home nav"
                  to="/home"
>
                  HOME
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="artists nav"
                  to="/artists"
>
                  ARTISTS
</NavLink>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "HOME",
                  "className": "home nav",
                  "to": "/home",
                },
                "ref": null,
                "rendered": "HOME",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "ARTISTS",
                  "className": "artists nav",
                  "to": "/artists",
                },
                "ref": null,
                "rendered": "ARTISTS",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/home"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/artists"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/home",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/artists",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="above-header"
>
            <h1>
                        Bands 
                        <span>
                                    in the
                        </span>
                         Backyard
            </h1>
            <div>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="home nav"
                                    to="/home"
                        >
                                    HOME
                        </NavLink>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="artists nav"
                                    to="/artists"
                        >
                                    ARTISTS
                        </NavLink>
            </div>
</div>,
          <header />,
          <div>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/home"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/artists"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Bands 
                <span>
                                in the
                </span>
                 Backyard
</h1>,
              <div>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="home nav"
                                to="/home"
                >
                                HOME
                </NavLink>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="artists nav"
                                to="/artists"
                >
                                ARTISTS
                </NavLink>
</div>,
            ],
            "className": "above-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Bands ",
                  <span>
                    in the
</span>,
                  " Backyard",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Bands ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "in the",
                  },
                  "ref": null,
                  "rendered": "in the",
                  "type": "span",
                },
                " Backyard",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="home nav"
                    to="/home"
>
                    HOME
</NavLink>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="artists nav"
                    to="/artists"
>
                    ARTISTS
</NavLink>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "HOME",
                    "className": "home nav",
                    "to": "/home",
                  },
                  "ref": null,
                  "rendered": "HOME",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "ARTISTS",
                    "className": "artists nav",
                    "to": "/artists",
                  },
                  "ref": null,
                  "rendered": "ARTISTS",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/home"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/artists"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/home",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/artists",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
